public/javascripts/main.js

```diff
@@ -123,14 +177,40 @@ const TopLevelBox = React.createClass({
             }.bind(this)
         });
     },
+    render(){
+        return(
+            <div>
+                <ArticlesBox articles={this.state.articles} articleFilter={this.props.articleFilter} />
+                <ModalNewArticleBox showModal={this.props.showModal} closeModal={this.props.closeModal} postNewArticle={this.postNewArticle}/>
+            </div>
+        );
+    }
+});
+
+const TopLevelBox = React.createClass({
+    getInitialState() {
+        return {
+            articleFilter: "",
+            showModal: false
+        };
+    },
+    openModal(){
+        this.setState({showModal: true});
+    },
+    closeModal(){
+        this.setState({showModal: false});
+    },
+    handleArticleFilterChange(event) {
```

Comment:
could call `/api/articles?title=${event.target.value}` and replace the results in the state to use server-side filtering out of the box.